engine/engine.go

```diff
@@ -160,7 +167,7 @@ func Start(ctx context.Context, startOpts *Config, fn StartCallback) error {
 				return nil, err
 			}
 
-			return bkgw.NewResult(), nil
+			return gwClient.CombinedResult(), nil
```

Comment:
From what I gather, the idea here is now we return _all_ results ever produced by the Dagger session so we can export everything? Neat!

Reply:
Yep (will add a comment here too), it's specifically everything that we call solve on.

Actually as I typed that out I realized that because we do exports in separate sessions the cache export for those will happen synchronously as part of the publish/export call. Not super comprehensible behavior but will leave as is for now, probably not worth blocking this first iteration over.

Also in general all of this behavior is going to start out very leaky in that to explain it to users we'd have to explain all sorts of buildkit internal concepts. The idea in the longer term is that we can make the decision on which results more intelligent (i.e. based on metrics around the execution time, layer size, backend config, etc.). That will also detach the explanation of what is + isn't cached from buildkit internals.